tensorboard/plugins/image/images_plugin.py

```diff
@@ -353,12 +412,16 @@ def _get_individual_image(self, run, tag, index, sample):
     @wrappers.Request.application
     def _serve_individual_image(self, request):
         """Serves an individual image."""
-        run = request.args.get("run")
-        tag = request.args.get("tag")
-        index = int(request.args.get("index", "0"))
-        sample = int(request.args.get("sample", "0"))
         try:
-            data = self._get_individual_image(run, tag, index, sample)
+            if self._data_provider:
+                blob_key = request.args["blob_key"]
+                data = self._data_provider.read_blob(blob_key)
+            else:
+                run = request.args.get("run")
+                tag = request.args.get("tag")
+                index = int(request.args.get("index", "0"))
+                sample = int(request.args.get("sample", "0"))
+                data = self._get_individual_image(run, tag, index, sample)
```

Comment:
`_get_individual_image()` internally uses either DB connection or Multiplexer, but do we also need to add a DataProvider path there?

Reply:
Yeah, this API is slightly wonky because the query parameters are
different in the two paths: the multiplexer/DB paths take run, tag,
index, and sample, which are interpreted by this plugin; but the generic
data path takes an opaque blob key, which is interpreted only by the
data provider. Python doesn’t have a nice way to express union types, so
I didn’t see a clean way for `_get_individaul_image` to handle this. But
for clarity I’ve gone ahead and factored this out into a method. Does
this help?

Reply:
Sorry, I didn't realize earlier that the data provider path doesn't use `_get_individual_image()`.  The rename to `_get_legacy_individual_image()` looks clearer to me.

For `_get_generic_data_individual_image`, feel free to inline it as before, if you prefer.